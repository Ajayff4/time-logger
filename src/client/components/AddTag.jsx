import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import axios from 'axios'
import { connect } from 'react-redux'
import { addTag } from '../actions'
import PropTypes from 'prop-types'

class AddTag extends Component {
    constructor(props) {
        super(props)
        this.state = {
            tag: "",
            category: "",
            categories: {}
        }
    }

    goToAddLogs = () => {
        this.props.history.push("/add-log")
    }

    componentDidMount() {
        if (this.state.categories !== {}) {
            axios.post("/categories")
                .then(res => {
                    this.setState({ categories: { ...res.data } })
                }).catch(err => {
                    console.error("post cat: ", err)
                })
        }

    }

    onChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    }

    onSubmit = (e) => {
        e.preventDefault()
        const { tag, category } = this.state
        axios.post('/add-tag', {
            tag: tag,
            category: category
        }).then(res => {
            console.log(res.status)
            this.props.addTag()
            this.props.history.push('/add-log')
        }).catch(err => {
            console.error("post: ", err)
        })
    }

    createCategory = () => {
        this.props.history.push("/add-category")
    }

    render() {
        return (
            <>
                <h2>Add Tag Component</h2>
                <center>
                    <div id="form-css">
                        <form method="POST" onSubmit={this.onSubmit}>
                            <input
                                type="text"
                                name="tag"
                                id="tag"
                                placeholder="Input tag name"
                                value={this.state.tag}
                                onChange={this.onChange}
                            /><br />
                            <select
                                name="category"
                                value={this.state.category}
                                onChange={this.onChange}
                            >
                                <option key="choose" value="--">Choose category</option>
                                {
                                    Object.entries(this.state.categories).map((category) => {
                                        category = category[1]["category"]
                                        return <option key={category} value={category}>{category}</option>
                                    })
                                }
                            </select>
                            <br />
                            <button type="submit" value="submit">Add Tag</button>
                            <br />
                            {/* eslint-disable-next-line */}
                            <a href="#" onClick={this.createCategory}>Create a new category</a>
                            <br />
                            {/* eslint-disable-next-line */}
                            <a href="#" onClick={this.goToAddLogs}>back to add log</a>
                        </form>
                    </div>
                </center>
            </>
        )
    }
}

AddTag.propTypes = {
    addTag: PropTypes.func.isRequired
}


const mapDispatchToProps = dispatch => {
    return {
        addTag: () => dispatch(addTag())
    }
}

export default connect(null, mapDispatchToProps)(withRouter(AddTag))