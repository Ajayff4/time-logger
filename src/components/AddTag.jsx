import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import axios from 'axios'
import { connect } from 'react-redux'
import * as actions from '../actions'
import PropTypes from 'prop-types'

class AddTag extends Component {
    static propTypes = {
        addTag: PropTypes.func.isRequired,
        failedToFetchData: PropTypes.func.isRequired
    }

    constructor(props) {
        super(props)
        this.state = {
            tag: "",
            category: "",
            categories: []
        }
    }

    goToAddLogs = () => {
        this.props.history.push("/add-log")
    }

    componentDidMount() {
        if (this.state.categories !== {}) {
            axios.get('http://localhost:5000/rest/category')
            .then(res => {
                this.setState({ categories: [...res.data] })

            }).catch(err => {
                console.log("Error in fetching logs")
            })
        }
    }

    onChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    }

    onSubmit = (e) => {
        e.preventDefault()
        const { tag, category } = this.state
        axios.post('http://localhost:5000/rest/tags', {
            tag: tag,
            category: category
        })
            .then(res => {
                this.props.addTag()
                this.props.history.push('/add-log')
            }).catch(err => {
                this.props.failedToFetchData("Error in adding tag")
            })
    }

    createCategory = () => {
        this.props.history.push("/add-category")
    }

    render() {
        return (
            <>
                <h2>Add Tag Component</h2>
                <center>
                    <div id="form-css">
                        <form method="POST" onSubmit={this.onSubmit}>
                            <input
                                type="text"
                                name="tag"
                                id="tag"
                                placeholder="Input tag name"
                                value={this.state.tag}
                                onChange={this.onChange}
                                required
                            /><br />
                            <select
                                name="category"
                                value={this.state.category}
                                onChange={this.onChange}
                                required
                            >
                                <option key="choose" value="--">Choose category</option>
                                {
                                    Object.entries(this.state.categories).map((category) => {
                                        category = category[1]["category"]
                                        return <option key={category} value={category}>{category}</option>
                                    })
                                }
                            </select>
                            <br />
                            <button type="submit" value="submit">Add Tag</button>
                            <br />
                            {/* eslint-disable-next-line */}
                            <a href="#" onClick={this.createCategory}>Create a new category</a>
                            <br />
                            {/* eslint-disable-next-line */}
                            <a href="#" onClick={this.goToAddLogs}>back to add log</a>
                        </form>
                    </div>
                </center>
            </>
        )
    }
}

const mapDispatchToProps = dispatch => {
    return {
        failedToFetchData: (error) => dispatch(actions.failedToFetchData(error)),
        getAllCategory: () => dispatch(actions.getAllCategory()),
        addTag: () => dispatch(actions.addTag())
    }
}

export default connect(null, mapDispatchToProps)(withRouter(AddTag))