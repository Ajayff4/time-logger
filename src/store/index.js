import { createStore, applyMiddleware } from 'redux'
import rootReducer from '../reducers/index'
//import logger from 'redux-logger'
import { composeWithDevTools } from 'redux-devtools-extension'
import rootSaga from '../saga'
import createSagaMiddleware from 'redux-saga'

const sagaMiddleware = createSagaMiddleware()
const store = createStore(
    rootReducer,
    composeWithDevTools(
        applyMiddleware(sagaMiddleware))
)

sagaMiddleware.run(rootSaga)

export default store