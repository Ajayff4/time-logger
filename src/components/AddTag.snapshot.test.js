import React from 'react'
import renderer from 'react-test-renderer'
import AddTag from './AddTag'
import { Provider } from 'react-redux';
import store from '../store';
import { BrowserRouter as Router } from 'react-router-dom';

describe('AddTag Snapshot Testing...', () => {
    it('renders component correctly', () => {
        const tree = renderer.create(<Provider store={store}><Router><AddTag /></Router></Provider> ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});