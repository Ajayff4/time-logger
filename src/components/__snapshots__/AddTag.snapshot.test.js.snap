// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTag Snapshot Testing... renders component correctly 1`] = `
Array [
  <h2>
    Add Tag Component
  </h2>,
  <center>
    <div
      id="form-css"
    >
      <form
        method="POST"
        onSubmit={[Function]}
      >
        <input
          id="tag"
          name="tag"
          onChange={[Function]}
          placeholder="Input tag name"
          required={true}
          type="text"
          value=""
        />
        <br />
        <select
          name="category"
          onChange={[Function]}
          required={true}
          value=""
        >
          <option
            value="--"
          >
            Choose category
          </option>
        </select>
        <br />
        <button
          type="submit"
          value="submit"
        >
          Add Tag
        </button>
        <br />
        <a
          href="#"
          onClick={[Function]}
        >
          Create a new category
        </a>
        <br />
        <a
          href="#"
          onClick={[Function]}
        >
          back to add log
        </a>
      </form>
    </div>
  </center>,
]
`;
